### ISSUE TITLE
"Execute after Success" for credit card

### ISSUE THREAD

**Issue by masudewucse** [paypal/PayPal-PHP-SDK#873]
Dear,
Right now i am integrating Paypal-Plus with my custom made PHP web shop. 
"PayPal_PLUS_Demoshop" shows a step 'Execute after Success' which is only available (in PayPal-PHP-SDK)  for 'Paypal' payment but not for 'Direct debit' or 'Master card' payment. 
Can you please help on that?
I would appreciate your help!

**Comment by xiaoleih41** [comment]
PayPal payments needs account holder's approval for execution while credit card payments doesn't require. Payments for credit card should be instant. Are you trying to do auth/capture? Please explain what you are trying to achieve for us better assisting you.

**Comment by masudewucse** [comment]
Please go [this](https://www.paypalobjects.com/digitalassets/c/website/marketing/emea/de/de/paypal-plus-center/PayPal_PLUS_Demoshop.pdf) pdf provided by paypal. 
(in the pdf) 
If you select credit card -> paypal website -> ask for credentials -> it comes back to the web-shop and ask for final confirmation as like Paypal (same for Direct dabit).

**Comment by xiaoleih41** [comment]
The credit card payment payment method was just a mask on the iframe apparently. The only valid payment resource in PayPal plus is PayPal account payments based on the [integration guide](https://www.paypalobjects.com/webstatic/de_DE/downloads/PayPal-PLUS-IntegrationGuide.pdf). Therefore you should execute the payment as it's a PayPal payment.

**Comment by masudewucse** [comment]
I think i was not able to make you clear about my question. let me try again.
I am using PayPal-PHP-SDK for Paypal plus integration.
According to PayPal-PHP-SDK: 
(From the Paypal plus i frame)
For Paypal payment: I click on "Paypal" ->Paypal website -> ask for Paypal credentials -> payment success-> redirects to web shop-> ask for the Final confirmation(execution after success) -> DONE 

For Credit Card : I click on "Credit Card" ->Paypal website -> ask forCredit Card credentials -> payment success-> redirects to web shop-> DONE
You see Credit card payment does not have "Execute after Success" step. 
But  Paypal Plus DEMO shop shows  "Execute after Success" for all Paypal, Credit card and direct Debit.

**Comment by masudewucse** [comment]
Ok cool.
So u meant All paypal plus payment options required "execution after Success".
Please dont close this ticket. I will let u know tomorrow. Thanks

**Comment by xiaoleih41** [comment]
Yes, please let me  know.

**Comment by masudewucse** [comment]
Yes i have checked and you are right! 
But the problem is with the Direct Debit payment, it suppose to be working as like as paypal and credit card.  In sandbox, when i try some test German bank details that always says 
<pre>Sorry, we can’t complete your purchase at this time
Please return to the merchant and choose another way to pay.</pre>
or
Or redirect to fail transaction url. 
Any clue?

**Comment by xiaoleih41** [comment]
Do you have a `debugId` that you can share?

**Comment by masudewucse** [comment]
when i try this fake bank IBAN: DE92123456789876543210 (Germany)
It redirects me to `payment failed` page `:payment.php?success=false&token=EC-7PV12205Y8145241R&ppp_msg=1`

**Comment by xiaoleih41** [comment]
It may have something to do with the bank IBAN number. Please consider to contact [PayPal technical support](www.paypal.com/mts). The regional tech support team should be able to help you with this question.

**Comment by masudewucse** [comment]
ok. yes i had a phone conversation with paypal support. they said, direct debit payment is not enabled for paypal sandbox and this will work in live.

**Comment by xiaoleih41** [comment]
That makes sense!  Appreciate this info. I am closing this ticket as the issue should be solved.
